txnsync/txngroups_test.go

```diff
@@ -363,30 +366,72 @@ func TestTxnGroupEncodingLarge(t *testing.T) {
 func BenchmarkTxnGroupEncoding(b *testing.B) {
 	txnGroups, _, _, err := txnGroupsData()
 	require.NoError(b, err)
-	var encodedGroupsBytes []byte
+	var size int
+	var s syncState
 
 	b.ReportAllocs()
 	b.ResetTimer()
 	for i := 0; i < b.N; i++ {
-		encodedGroupsBytes, err := encodeTransactionGroups(txnGroups)
+		encodedGroupsBytes, _, err := s.encodeTransactionGroups(txnGroups, 1000000000)
 		require.NoError(b, err)
+		size = len(encodedGroupsBytes)
 		releaseEncodedTransactionGroups(encodedGroupsBytes)
 	}
 
-	fmt.Println("new data: ", len(encodedGroupsBytes))
+	b.ReportMetric(float64(size), "encodedDataBytes")
+}
+
+func BenchmarkTxnGroupCompression(b *testing.B) {
+	txnGroups, _, _, err := txnGroupsData()
+	require.NoError(b, err)
+	var size int
+	var s syncState
+	encodedGroupsBytes, compressionFromat, err := s.encodeTransactionGroups(txnGroups, 1000000000)
+	require.NoError(b, err)
+	require.Equal(b, compressionFromat, compressionFormatNone)
+
+	b.ReportAllocs()
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		compressedGroupBytes, err := compressTransactionGroupsBytes(encodedGroupsBytes)
+		require.NoError(b, err)
+		size = len(compressedGroupBytes)
+	}
+
+	b.ReportMetric(float64(size), "encodedDataBytes")
```

Comment:
This benchmark would need to "generate" two metrics:
1. The estimatedGzipCompressionSpeed, by `len(encodedGroupsBytes)/loop time`
2. The estimatedGzipCompressionGains, by `( len(encodedGroupsBytes)-size) / len(encodedGroupsBytes)`

These would allow us to calibrate the above parameters when we flip the compression library.

Reply:
Is there a convenient way to access loop time?

Reply:
Not when using the go benchmarks. You can make this method into a *TestXXX* and then use the `testing.Benchmark` method to get the `BenchmarkResult` and print it out. But that's already more than really needed. Just wrap the for loop in a `loopStartTime := time.Now()... loopDuration := time.Now().Sub(loopStartTime)` and you'll have what you need.